File: java/parser/src/main/java/gov/lbl/parser/parser/Expression_listVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Expression;
import gov.lbl.parser.domain.Expression_list;

import static java.util.stream.Collectors.toList;

import java.util.List;

public class Expression_listVisitor  extends modelicaBaseVisitor<Expression_list> {
    @Override
    public Expression_list visitExpression_list(modelicaParser.Expression_listContext ctx) {
        ExpressionVisitor expressionVisitor = new ExpressionVisitor();
        List<Expression> expressions = ctx.expression() == null ? null : ctx.expression()
                                                                                    .stream()
                                                                                    .map(expression -> expression.accept(expressionVisitor))
                                                                                    .collect(toList());
        return new Expression_list(expressions);
    }
}

File: jsParser/parser/Expression_listVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Expression_list = require('../domain/Expression_list')

const ExpressionVisitor = require('./ExpressionVisitor')

class Expression_listVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitExpression_list (ctx) {
    const expressions = []

    if (ctx.expression()) {
      const expressionVisitor = new ExpressionVisitor.ExpressionVisitor()
      ctx.expression().forEach(expr => {
        expressions.push(expressionVisitor.visitExpression(expr))
      })
    }
    return new Expression_list.Expression_list(expressions)
  }
};

Expression_listVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.Expression_listVisitor = Expression_listVisitor
